addons/sale_mrp/models/mrp_bom.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class MrpBom(models.Model):
+    _inherit = 'mrp.bom'
+
+    def toggle_active(self):
+        self.filtered(lambda bom: bom.active)._ensure_bom_is_free()
+        return super().toggle_active()
+
+    def unlink(self):
+        self._ensure_bom_is_free()
+        return super().unlink()
+
+    def _ensure_bom_is_free(self):
+        product_ids = []
+        for bom in self:
+            if bom.type != 'phantom':
+                continue
+            product_ids += bom.product_id.ids or bom.product_tmpl_id.product_variant_ids.ids
+        if not product_ids:
+            return
+        lines = self.env['sale.order.line'].search([
```

Comment:
maybe add sudo() and limit=1 to prevent any issue (some user have no access to sale data, and prevent a memory error).

@Whenrow